addon/model.js

```diff
@@ -98,33 +112,76 @@ class YesManAttributesSingletonClass {
 
 const YesManAttributes = new YesManAttributesSingletonClass();
 
-const retrieveFromCurrentState = computed('_topModel.currentState', function(key) {
-  return this._topModel._internalModel.currentState[key];
-}).readOnly();
-
 // global buffer for initial properties to work around
 //  a)  can't write to `this` before `super`
 //  b)  core_object writes properties before calling `init`; this means that no
 //      CP or setknownProperty can rely on any initialization
 let initProperites = Object.create(null);
 
+let ignoreInvalidRequestsMap = new WeakMap();
+let ignoreErrorRequestsMap = new WeakMap();
 export default class MegamorphicModel extends EmberObject {
   init(properties) {
     // Drop Ember.Object subclassing instead
     super.init(...arguments);
+    if (false) {
+      recordDataToRecordMap.set(properties._recordData, this);
+      this._recordData = properties._recordData;
+      this._invalidRequests = [];
+      this._errorRequests = [];
```

Comment:
Let's add comments explaining why these dual arrays are actually necessary or, if they are not, remove them.

This is more complicated than I would expect.

Reply:
Added a comment saying that they mimic the current ED implementation. I will try removing them there, and then if that works, do the same thing here.

Reply:
@igort how did that go?